*Request.HeaderPresent not working in v1.0.16

I have Test Code like this, 

```
func TestHitHTTP(t *testing.T) {
	// create custom HTTP client
	httpClient := &http.Client{
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{
				InsecureSkipVerify: true,
			},
		},
	}

	// define URL
	URL := "https://ayssiz.com"

	// define mock
	gock.InterceptClient(httpClient)
	defer gock.Off()
	defer gock.RestoreClient(httpClient)

	// create mock response
	mockResponse := make(map[string]interface{})
	mockResponse["success"] = true

	gock.New(URL).
		HeaderPresent("KEY").
		Reply(200).
		JSON(mockResponse)

	// hit HTTP
	hitHttp(httpClient, URL)

	assert.Equal(t, true, gock.IsDone())
}


func hitHttp(httpClient *http.Client, URL string) (interface{}, error) {
	var jsonBody interface{}

	request, err := http.NewRequest("POST", URL, nil)
	request.Header.Set("KEY", "APIKEY")

	response, err := httpClient.Do(request)
	if err != nil {
		return jsonBody, err
	}

	defer response.Body.Close()
	json.NewDecoder(response.Body).Decode(&jsonBody)

	return jsonBody, nil
}
```



I use HeaderPresent to check wheter the header key is present or not. Previously it works fine in v1.0.15, 
but when I upgrade it to v1.0.16, it is not working. The value of gock.IsDone() is false, which mean the mock is not hit by the httpClient


--- FAIL: TestHitHTTP (0.00s)
    module_test.go:43: 
                Error Trace:    module_test.go:43
                Error:          Not equal: 
                                expected: true
                                actual  : false
                Test:           TestHitHTTP
FAIL

--- a/matchers.go
+++ b/matchers.go
@@ -81,7 +81,7 @@ func MatchHeaders(req *http.Request, ereq *Request) (bool, error) {
 
 		for _, field := range req.Header[key] {
 			match, err = regexp.MatchString(value[0], field)
-			//Some values may contain reserved regex params e.g. "()", try matching with these escaped
+			// Some values may contain reserved regex params e.g. "()", try matching with these escaped.
 			matchEscaped, err = regexp.MatchString(regexp.QuoteMeta(value[0]), field)
 
 			if err != nil {

--- a/request.go
+++ b/request.go
@@ -194,20 +194,20 @@ func (r *Request) BasicAuth(username, password string) *Request {
 
 // MatchHeader defines a new key and value header to match.
 func (r *Request) MatchHeader(key, value string) *Request {
-	r.Header[key] = []string{value}
+	r.Header.Set(key, value)
 	return r
 }
 
 // HeaderPresent defines that a header field must be present in the request.
 func (r *Request) HeaderPresent(key string) *Request {
-	r.Header[key] = []string{".*"}
+	r.Header.Set(key, ".*")
 	return r
 }
 
 // MatchHeaders defines a map of key-value headers to match.
 func (r *Request) MatchHeaders(headers map[string]string) *Request {
 	for key, value := range headers {
-		r.Header[key] = []string{value}
+		r.Header.Set(key, value)
 	}
 	return r
 }

--- a/request_test.go
+++ b/request_test.go
@@ -15,7 +15,7 @@ func TestNewRequest(t *testing.T) {
 	st.Expect(t, req.URLStruct.Host, "foo.com")
 	st.Expect(t, req.URLStruct.Scheme, "http")
 	req.MatchHeader("foo", "bar")
-	st.Expect(t, req.Header["foo"][0], "bar")
+	st.Expect(t, req.Header.Get("foo"), "bar")
 }
 
 func TestRequestSetURL(t *testing.T) {
@@ -97,10 +97,10 @@ func TestRequestMatchHeader(t *testing.T) {
 	req.MatchHeader("UPPERCASE", "bat")
 	req.MatchHeader("Mixed-CASE", "foo")
 
-	st.Expect(t, req.Header["foo"][0], "bar")
-	st.Expect(t, req.Header["bar"][0], "baz")
-	st.Expect(t, req.Header["UPPERCASE"][0], "bat")
-	st.Expect(t, req.Header["Mixed-CASE"][0], "foo")
+	st.Expect(t, req.Header.Get("foo"), "bar")
+	st.Expect(t, req.Header.Get("bar"), "baz")
+	st.Expect(t, req.Header.Get("UPPERCASE"), "bat")
+	st.Expect(t, req.Header.Get("Mixed-CASE"), "foo")
 }
 
 func TestRequestHeaderPresent(t *testing.T) {
@@ -109,10 +109,10 @@ func TestRequestHeaderPresent(t *testing.T) {
 	req.HeaderPresent("bar")
 	req.HeaderPresent("UPPERCASE")
 	req.HeaderPresent("Mixed-CASE")
-	st.Expect(t, req.Header["foo"][0], ".*")
-	st.Expect(t, req.Header["bar"][0], ".*")
-	st.Expect(t, req.Header["UPPERCASE"][0], ".*")
-	st.Expect(t, req.Header["Mixed-CASE"][0], ".*")
+	st.Expect(t, req.Header.Get("foo"), ".*")
+	st.Expect(t, req.Header.Get("bar"), ".*")
+	st.Expect(t, req.Header.Get("UPPERCASE"), ".*")
+	st.Expect(t, req.Header.Get("Mixed-CASE"), ".*")
 }
 
 func TestRequestMatchParam(t *testing.T) {

